airbyte-tests/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java

```diff
@@ -0,0 +1,276 @@
+                              final Map<String, String> env,
+                              final Map<String, Consumer<String>> customServiceLogListeners) {
+    this.dockerComposeFile = dockerComposeFile;
+    this.env = env;
+    this.customServiceLogListeners = customServiceLogListeners;
+  }
+
+  /**
+   * Starts Airbyte docker-compose configuration. Will block until the server is reachable or it times
+   * outs.
+   */
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public void start() throws IOException, InterruptedException {
+    final File cleanedDockerComposeFile = prepareDockerComposeFile(dockerComposeFile);
+    dockerComposeContainer = new DockerComposeContainer(cleanedDockerComposeFile).withEnv(env);
+    serviceLogConsumer(dockerComposeContainer, "init");
+    serviceLogConsumer(dockerComposeContainer, "db");
+    serviceLogConsumer(dockerComposeContainer, "seed");
+    serviceLogConsumer(dockerComposeContainer, "scheduler");
+    serviceLogConsumer(dockerComposeContainer, "server");
+    serviceLogConsumer(dockerComposeContainer, "webapp");
+    serviceLogConsumer(dockerComposeContainer, "airbyte-temporal");
+
+    dockerComposeContainer.start();
+
+    waitForAirbyte();
+  }
+
+  private static Map<String, String> prepareDockerComposeEnvVariables(File envFile) throws IOException {
+    LOGGER.info("Searching for environment in {}", envFile);
+    Preconditions.checkArgument(envFile.exists(), "could not find docker compose environment");
+
+    final Properties prop = new Properties();
+    prop.load(new FileInputStream(envFile));
+    return Maps.fromProperties(prop);
+  }
+
+  /**
+   * TestContainers docker compose files cannot have container_names, so we filter them.
+   */
+  private static File prepareDockerComposeFile(File originalDockerComposeFile) throws IOException {
+    final File cleanedDockerComposeFile = Files.createTempFile(Path.of("/tmp"), "docker_compose", "acceptance_test").toFile();
+
+    try (final Scanner scanner = new Scanner(originalDockerComposeFile)) {
+      try (final FileWriter fileWriter = new FileWriter(cleanedDockerComposeFile)) {
+        while (scanner.hasNextLine()) {
+          final String s = scanner.nextLine();
+          if (s.contains("container_name")) {
+            continue;
+          }
+          fileWriter.write(s);
+          fileWriter.write('\n');
+        }
+      }
+    }
+    return cleanedDockerComposeFile;
+  }
+
+  @SuppressWarnings("BusyWait")
+  private static void waitForAirbyte() throws InterruptedException {
+    // todo (cgardens) - assumes port 8001 which is misleading since we can start airbyte on other
+    // ports. need to make this configurable.
+    final AirbyteApiClient apiClient = new AirbyteApiClient(
+        new ApiClient().setScheme("http")
+            .setHost("localhost")
+            .setPort(8001)
+            .setBasePath("/api"));
+
+    final HealthApi healthApi = apiClient.getHealthApi();
```

Comment:
this is pretty clever haha

Reply:
thanks! we had been using timeouts and then i realized we have an endpoint that handles this exact purpose. 🤣